dkist/dataset/tiled_dataset.py

```diff
@@ -124,5 +125,30 @@ def shape(self):
         """
         return self._data.shape
 
-    # TODO: def plot()
+    def plot(self, slice_index: int, share_zscale=False, **kwargs):
+        vmin, vmax = np.inf, 0
+        fig = plt.figure()
+        for i, tile in enumerate(self.flat):
+            ax = fig.add_subplot(self.shape[0], self.shape[1], i+1, projection=tile[0].wcs)
+            tile[slice_index].plot(axes=ax, **kwargs)
+            if i == 0:
+                xlabel = ax.coords[0].get_axislabel() or ax.coords[0]._get_default_axislabel()
+                ylabel = ax.coords[1].get_axislabel() or ax.coords[1]._get_default_axislabel()
+                for coord in ax.coords:
+                    if "b" in coord.axislabels.get_visible_axes():
+                        fig.supxlabel(xlabel, y=0.05)
+                    if "l" in coord.axislabels.get_visible_axes():
+                        fig.supylabel(ylabel, x=0.05)
+            axmin, axmax = ax.get_images()[0].get_clim()
+            vmin = axmin if axmin < vmin else vmin
+            vmax = axmax if axmax > vmax else vmax
+            ax.set_ylabel(" ")
+            ax.set_xlabel(" ")
+        if share_zscale:
+            for ax in fig.get_axes():
+                ax.get_images()[0].set_clim(vmin, vmax)
+        timestamp = self[0, 0].axis_world_coords("time")[-1].iso[slice_index]
+        fig.suptitle(f"TiledDataset {self.inventory['datasetId']} at time {timestamp} (slice={slice_index})", y=0.95)
```

Comment:
I think it would be better if this were "VBI Dataset (ASKJD) at time ..." if you don't mind.